### PR TITLE
Adding Xray Source Control Service

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/jfrog-cli/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.

---

**Adding XSC support to JFrog Cli and it's components**

- For users who have XSC enabled in their system, will refer to XSC server instead of Xray.
- Refactor to allow the future development of XSC service with fallback to Xray.
- XSC is build on-top of Xray service, meaning unimplemented XSC functions will fallback to default Xray service functions.
- For start, `jf-audit` command is implemented to work with XSC, sending extra GitInfoContext object to allow linking scan data to repositories.
- XSC url is not provided from flag or config. We build it by either taking the platform url and adding suffix, or replace Xray url suffix with `/xsc` suffix.

<h2>How to use </h2>

No action from the user is needed, as this is optional. 
If the server has XSC, it will query it.

In order to use the new functionality of linking scan information to repository, need to set the XscGitinfoContext object when calling `audit` command:
```
auditParams.xrayGraphScanParams.XscGitInfoContext = &scan.XscGitInfoContext{
		GitRepoUrl:    "https://github.com/EyalDelarea/npmexample",
		GitRepoName:   "npmexample",
		GitProject:    "myproject",
		GitProvider:   "github",
		BranchName:    "main",
		LastCommit:    "https://github.com/EyalDelarea/npmexample/commit/02c4eb4065e293d96fdfc70e64a58244eabed399",
		CommitHash:    "02c4eb4065e293d96fdfc70e64a58244eabed399",
		CommitMessage: "test dev deps",
		CommitAuthor:  "eyaldelarea",
		Date:         "2006-01-02T15:04:05Z07:00",
	}
```

Depends on 
JFrog client go https://github.com/jfrog/jfrog-client-go/pull/811
JFrog cli-core https://github.com/jfrog/jfrog-cli-core/pull/896

### CODE REVIEW

File: .github/workflows/xrayTests.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        suite: [ xray,xsc ]
```
Comment (sverdlov93) [inline .github/workflows/xrayTests.yml:22]:
```suggestion
        suite: [ xray, xsc ]
```

Comment (sverdlov93) [reply .github/workflows/xrayTests.yml:22]: also, this is not doing anything
you should change 
`--test.xray` to be `--test.${{ matrix.suite }}`

File: xray_test.go [hunk 2]
```diff
@@ -138,16 +139,19 @@ func TestXrayBinaryScanWithBypassArchiveLimits(t *testing.T) {
...
+	output := testXrayAuditNpm(t, string(utils.SimpleJson), false)
 	verifySimpleJsonScanResults(t, output, 1, 1)
 }
 
-func testXrayAuditNpm(t *testing.T, format string) string {
+func testXrayAuditNpm(t *testing.T, format string, isXsc bool) string {
+	if !isXsc {
+		initXrayTest(t, scangraph.GraphScanMinXrayVersion)
+	}
 	initXrayTest(t, scangraph.GraphScanMinXrayVersion)
```
Comment (sverdlov93) [inline xray_test.go:155]: ?

File: xsc_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	}
+	validateXscVersion(t, minVersion)
+}
+func validateXscVersion(t *testing.T, minVersion string) {
+	err := clientutils.ValidateMinimumVersion(clientutils.Xray, xrayDetails.XscVersion, minVersion)
+	if err != nil {
+		t.Skip(err)
+	}
+}
+func TestXSCAudit(t *testing.T) {
```
Comment (sverdlov93) [inline xsc_test.go:22]:
```suggestion
func TestXscAudit(t *testing.T) {
```

File: xsc_test.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	initXscTest(t, scangraph.XscMinVersion)
+	testXrayAuditNpm(t, "json", true)
```
Comment (sverdlov93) [inline xsc_test.go:24]:
```suggestion
	testXrayAuditNpm(t, string(utils.Json), true)
```

**Comment by EyalDelarea** [comment]
XSC shouldn't be tested via the CLI at this point.

**Pull request closed without merging**